### ISSUE TITLE
builtinfun.length undefined

### ISSUE THREAD

**Issue by thiagoarrais** [boa-dev/boa#210]
I'm getting `undefined` when I do `'abc'.matchAll.length`. This is incompatible with Chrome's implementation (for example), where I get `1`.

Testing with master with the following `tests.js`:

```js
console.log('abc'.matchAll.length);
```

Here's the output for `cargo run`:

```
$ cargo run
   Compiling Boa v0.5.0 (/data/boa)
    Finished dev [unoptimized + debuginfo] target(s) in 3.20s
     Running `target/debug/boa`
undefined
[src/bin/bin.rs:57] exec(&buffer) = "undefined"
```

Related to #193 and #206

**Comment by Stupremee** [comment]
I think I found the error. 
The following code:
```js
'abc'.matchAll
```
returns undefined and not the function object.
Firefox console returns the function object instead.
I will investigate further and try to find the cause

**Comment by Stupremee** [comment]
I found the error
If you create a string with `"` or `'`, the `builtins::string::create_constructor` (aka String Constructor) is not used, and so the methods are not set.
When a method is called from this string, this string is first converted to an object by `Interpreter::to_object`, this `to_object` method then uses the string prototype which has all methods set etc, so the `matchAll` method is also a property in the object. However, there is another problem. If the object you want to get a field from is a function, the `to_obj` method returns an error. I'm not sure how best to fix the error. At the moment I can only offer the following Quick-Fix:
Change [this](https://github.com/jasonwilliams/boa/blob/master/src/lib/exec.rs#L106-L109)
to this:
```rust
ExprDef::GetConstField(ref obj, ref field) => {
    let mut obj = self.run(obj)?;
    if obj.get_type() != "function"
        && (obj.get_type() != "object" || obj.get_type() != "symbol")
    {
        obj = self.to_object(&obj).expect("failed to convert to object");
    }
    Ok(obj.borrow().get_field_slice(field))
}
```

**Comment by Stupremee** [comment]
Is there now a good solution which should be used?

**Comment by jasonwilliams** [comment]
Hey @Stupremee do you think this logic should be inside value.rs on `get_field` ?

**Comment by Stupremee** [comment]
I think so yes

**Comment by Stupremee** [comment]
I can take care of this issue. I will add the logic in `get_value`. If I find a better solution I will ask it here

**Comment by Stupremee** [comment]
I thought about this issue again, and I think it's the best method to create constant values (string, numbers) via the specific constructor.

**Comment by Razican** [comment]
I think this particular case could be solved by function objects, in #255.

**Comment by jasonwilliams** [comment]
I agree

**Comment by Razican** [comment]
What is the state of this, @jasonwilliams ?

**Comment by jasonwilliams** [comment]
This should be easy to add now we have function objects, il give this issue a v0.9 milestone

**Comment by croraf** [comment]
The original problem is solved but 
```js
'asd'['matchAll'].length
```
isn't.

There is a missmatch here: https://github.com/boa-dev/boa/blob/master/boa/src/exec/field/mod.rs#L10-L14

**Comment by jasonwilliams** [comment]
@croraf https://github.com/boa-dev/boa/pull/469 Doesn’t really close this issue, it just fixes and tests bracket accessing.
There’s no tests in there for length on built ins

**Comment by croraf** [comment]
@jasonwilliams I intentionally didn't included this test, as it follows from the two tests that I did include.

`'string'.builtinFunction.length` is just a combination of 
 - `'string'.builtinFunction` resolution and 
 - `function.property` resolution

Both of these were failing, and now both are fixed which I checked with tests, so their combination also works.

**Comment by Razican** [comment]
I think this is fixed now, right? I can get the proper results using both methods.

I'm closing this for now, and removing the milestone, since it seems it was solved at least on 0.9.
